catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/MigrationFileWriter.java

```diff
@@ -0,0 +1,119 @@
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.MetacardImpl;
+
+/**
+ * Responsible for all low-level IO operations pertaining to migration.
+ */
+public class MigrationFileWriter {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MigrationFileWriter.class);
+
+    //      ========================================================
+    //      TODO: Look into moving these functions to MigratableUtil
+    //      ========================================================
+
+    /**
+     * Creates the directory for the catalog export if it doesn't exist.
+     * <p>
+     * If the directory does exist, work based on the assumption it was cleaned by the parent
+     * process in the management logic of inport and export. There is no point to delegate
+     * low level directory operations to each and every migratable when they can be abstracted
+     * into the base logic.
+     *
+     * @param exportPath The path representing the directory to create.
+     * @throws MigrationException thrown if the export directory doesn't exist and could not be created
+     */
+    public void init(Path exportPath) throws MigrationException {
+        final File exportDir = exportPath.toFile();
+        if (!exportDir.exists() || !exportDir.isDirectory()) {
+            if (!exportDir.mkdir()) {
+                throw new MigrationException("Could not create catalog directory at export location");
+            } else {
+                LOGGER.debug("Created export directory: 'catalog'");
+            }
+        }
+    }
+
+    /**
+     * Writes the metacards in the provided results list to the provided file.
+     *
+     * @param exportFile The file to write to.
+     * @param results    The metacards to write into the file.
+     */
+    public void writeMetacards(File exportFile, final List<Result> results) throws IOException {
+        if (!exportFile.exists()) {
+            exportFile.createNewFile();
+        }
+
+        try (
+                FileOutputStream fileStream = createFileStream(exportFile);
+                BufferedOutputStream bufferedStream = createBufferedStream(fileStream);
+                ObjectOutputStream objectStream = createObjectStream(bufferedStream);
+        ) {
+            for (final Result result : results) {
+                Metacard metacard = new MetacardImpl(result.getMetacard());
+                objectStream.writeObject(metacard);
+            }
+            bufferedStream.flush();
+        }
+    }
+
+    ObjectOutputStream createObjectStream(OutputStream source) throws IOException {
+        return new ObjectOutputStream(source);
+    }
+
+    BufferedOutputStream createBufferedStream(OutputStream source) throws IOException {
+        return new BufferedOutputStream(source);
+    }
+
+    FileOutputStream createFileStream(File file) throws IOException {
+        return new FileOutputStream(file);
+    }
+
+    /**
```

Comment:
Let's delete the commented out code for now. We can always save it some place else (e.g., local branch) if we want.